tests/test.rs

```diff
@@ -0,0 +1,270 @@
+    compare_f64(nom1.functions_min(), &nom2["functions_min"]);
+    compare_f64(nom1.functions_max(), &nom2["functions_max"]);
+    compare_f64(nom1.closures_min(), &nom2["closures_min"]);
+    compare_f64(nom1.closures_max(), &nom2["closures_max"]);
+
+    let mi1 = &metrics1.mi;
+    let mi2 = &metrics2["mi"];
+    compare_f64(mi1.mi_original(), &mi2["mi_original"]);
+    compare_f64(mi1.mi_sei(), &mi2["mi_sei"]);
+    compare_f64(mi1.mi_visual_studio(), &mi2["mi_visual_studio"]);
+
+    // Recursion
+    for (pos, s) in funcspace.spaces.iter().enumerate() {
+        compare_structs(s, &value["spaces"][pos]);
+    }
+}
+
+/// Compares two f64 values truncated to 3 decimals
+fn compare_f64(f1: f64, fv: &serde_json::Value) {
+    if f1.is_nan() || f1.is_infinite() {
+        assert!(fv.is_null());
+    } else {
+        let ft1 = f64::trunc(f1 * 1000.0) / 1000.0;
+        let ft2 = f64::trunc(fv.as_f64().unwrap() * 1000.0) / 1000.0;
+        assert!((ft1 - ft2).abs() < f64::EPSILON);
+    }
+}
+
+const OUTPUT_FOLDER: &str = "./tests/repositories/rca-output";
+
+/// Produces metrics runtime and compares them with previously generated json files
+fn compare_rca_output_with_files(
+    repo_branch: &str,
+    repo_url: &str,
+    repo_folder: &str,
+    include: &[&str],
+) {
+    // The first test clones the repository
+    // Next tests wait here until the repository is cloned
+    INIT.call_once(|| {
+        clone_repository(
+            "main",
+            "https://github.com/SoftengPoliTo/rca-output.git",
+            OUTPUT_FOLDER,
+        );
+    });
+
+    clone_repository(repo_branch, repo_url, repo_folder);
+
+    let num_jobs = 4;
+
+    let cfg = Config {
+        language: None,
+        output_folder: OUTPUT_FOLDER.to_owned(),
+    };
+
+    let mut gsbi = GlobSetBuilder::new();
+    for file in include {
+        gsbi.add(Glob::new(file).unwrap());
+    }
+
+    let files_data = FilesData {
+        include: gsbi.build().unwrap(),
+        exclude: GlobSet::empty(),
+        paths: vec![repo_folder.to_owned()],
+    };
+
+    if let Err(e) = ConcurrentRunner::new(num_jobs, act_on_file).run(cfg, files_data) {
+        eprintln!("{:?}", e);
+        process::exit(1);
+    }
+}
+
+/// Runs a git clone command
+fn clone_repository(branch: &str, url: &str, destination: &str) {
+    if !Path::new(destination).exists() {
+        Command::new("git")
+            .args([
+                "clone",
+                "--depth",
+                "1",
+                "--branch",
+                branch,
+                url,
+                destination,
+            ])
+            .output()
+            .expect("Git clone failed");
+    }
+}
+
+#[test]
+fn test_deepspeech() {
+    compare_rca_output_with_files(
+        "v0.9.3",
+        "https://github.com/mozilla/DeepSpeech.git",
+        "./tests/repositories/DeepSpeech",
+        &["*.cc", "*.cpp", "*.h", "*.hh"],
+    );
+}
+
+#[test]
+fn test_grcov() {
```

Comment:
We could use the Rust standard library ("library" subfolder in https://github.com/rust-lang/rust, but you can just use the whole repo since you have the `.rs` glob) for testing Rust, this way we make sure we're good with latest Rust language features (they are definitely going to use them in the Rust standard library).

Reply:
Thank you for the review! Do you think I should substitute grcov tests with Rust standard library tests? Or I can leave grcov tests and simply add Rust standard library tests?

Reply:
Both options work for me, you choose ;)